aderyn_core/src/detect/high/weak_randomness.rs

```diff
@@ -0,0 +1,190 @@
+use std::collections::BTreeMap;
+use std::error::Error;
+
+use crate::ast::{Expression, FunctionCall, FunctionCallKind, NodeID};
+
+use crate::capture;
+use crate::detect::detector::IssueDetectorNamePool;
+use crate::{
+    context::workspace_context::{ASTNode, WorkspaceContext},
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct WeakRandomnessDetector {
+    // Keys are: [0] source file name, [1] line number, [2] character location of node.
+    // Do not add items manually, use `capture!` to add nodes to this BTreeMap.
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for WeakRandomnessDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        let keccaks: Vec<&FunctionCall> = context.function_calls()
+            .into_iter()
+            .filter(|x| matches!(*x.expression, Expression::Identifier(ref id) if id.name == "keccak256"))
+            .collect();
+
+        for keccak in keccaks {
+            // keccak256 must have exactly one argument
+            let arg = keccak.arguments.get(0).unwrap();
+
+            if let Expression::FunctionCall(ref function_call) = *arg {
+                if check_encode(function_call) {
+                    capture!(self, context, keccak);
+                }
+            }
+            // get variable definition
+            else if let Expression::Identifier(ref i) = *arg {
+                if let Some(node_id) = i.referenced_declaration {
+                    let decleration = context.get_parent(node_id);
+
+                    if let Some(ASTNode::VariableDeclarationStatement(var)) = decleration {
+                        if let Some(Expression::FunctionCall(function_call)) = &var.initial_value {
```

Comment:
Can you please explain why use `get_parent` ? If we want to get the variable declaration we can do `context.nodes.get(node_id)` right?

Reply:
I need `VariableDeclarationStatement` because it already includes the `initial_value`. If I used `context.nodes.get(node_id)` I'd also need to get its parent, I think, resulting in more code.

But please tell me if you know an easier way to get the `initial_value`.